pertpy/plot/_milopy.py

```diff
@@ -0,0 +1,228 @@
+        """
+        try:
+            nhood_adata = mdata["milo_compositional"].T.copy()
+        except KeyError:
+            print(
+                "mdata should be a MuData object with two slots: 'rna' and 'milo_compositional' - please run milopy.count_nhoods(adata) first"
+            )
+
+        if subset_nhoods is not None:
+            nhood_adata = nhood_adata[subset_nhoods]
+
+        try:
+            nhood_adata.obs[anno_col]
+        except KeyError:
+            print(
+                'Cannot find {a} in adata.uns["nhood_adata"].obs -- \
+                    please run milopy.utils.annotate_nhoods(adata, anno_col) first'.format(
+                    a=anno_col
+                )
+            )
+
+        try:
+            nhood_adata.obs["logFC"]
+        except KeyError:
+            print(
+                'Cannot find `logFC` in adata.uns["nhood_adata"].obs -- \
+                    please run milopy.core.da_nhoods(adata) first'
+            )
+
+        sorted_annos = (
+            nhood_adata.obs[[anno_col, "logFC"]].groupby(anno_col).median().sort_values("logFC", ascending=True).index
+        )
+
+        anno_df = nhood_adata.obs[[anno_col, "logFC", "SpatialFDR"]].copy()
+        anno_df["is_signif"] = anno_df["SpatialFDR"] < alpha
+        # anno_df['across_organs'] = ["Significant across organs" if x else "" for x in (keep_nhoods & signif_nhoods)]
+        anno_df = anno_df[anno_df[anno_col] != "nan"]
+
+        try:
+            obs_col = nhood_adata.uns["annotation_obs"]
+            anno_palette = dict(zip(mdata["rna"].obs[obs_col].cat.categories, mdata["rna"].uns[f"{obs_col}_colors"]))
+            sns.violinplot(
+                data=anno_df,
+                y=anno_col,
+                x="logFC",
+                order=sorted_annos,
+                size=190,
+                inner=None,
+                orient="h",
+                palette=anno_palette,
+                linewidth=0,
+                scale="width",
+            )
+        except BaseException:
+            sns.violinplot(
+                data=anno_df,
+                y=anno_col,
+                x="logFC",
+                order=sorted_annos,
+                size=190,
+                inner=None,
+                orient="h",
+                linewidth=0,
+                scale="width",
+            )
+        sns.stripplot(
+            data=anno_df,
+            y=anno_col,
+            x="logFC",
+            order=sorted_annos,
+            size=2,
+            hue="is_signif",
+            palette=["grey", "black"],
+            orient="h",
+            alpha=0.5,
+        )
+        plt.legend(loc="upper left", title=f"< {int(alpha*100)}% SpatialFDR", bbox_to_anchor=(1, 1), frameon=False)
+        plt.axvline(x=0, ymin=0, ymax=1, color="black", linestyle="--")
+
+    @staticmethod
+    def nhood_counts_by_cond(mdata: MuData, test_var: str, subset_nhoods: list = None, log_counts: bool = False):
```

Comment:
This method doesn't work: it's still geared to have nhoods in obs and samples in var, as in old milopy implementation.

Reply:
Hi Emma, could you please explain what's the idea behind this function? At the moment I don't have clue of how to revise it to fit it into the current setting. Especially this part: 
```
if subset_nhoods is None:
    subset_nhoods = nhood_adata.obs_names

pl_df = pd.DataFrame(nhood_adata[subset_nhoods].X.A, columns=nhood_adata.var_names).melt(
    var_name=nhood_adata.uns["sample_col"], value_name="n_cells"
)
pl_df = pd.merge(pl_df, nhood_adata.var)
```

Thanks a lot!

Reply:
This function is to visualize the cell abundance for each sample, grouped by the covariate of interest. So it's a way to sanity check that the predicted enrichment/depletions make sense and what sort of effect size is captured.

Looking back at this the problem doesn't seem to be w the function, but it was complaining about the gastrulation dataset having integers in `mdata['milo_compositional'].obs` columns, so maybe it just needs to be a bit more robust (and to change the parameter names, cos `test_var` is actually a column in `mdata['milo_compositional'].obs`). To give you an ides of the output, on the tutorial dataset:
```
## Get ixs of enriched nhoods 
pl_nhoods = mdata['milo_compositional'].var_names[
    (mdata['milo_compositional'].var['SpatialFDR'] < 0.1) &
    (mdata['milo_compositional'].var['logFC'] > 0)
]

mdata['milo_compositional'].obs['sample'] = mdata['milo_compositional'].obs['sample'].astype('str')
mdata['milo_compositional'].obs['stage_continuous'] = mdata['milo_compositional'].obs['stage_continuous'].astype('category')
pt.pl.milo.nhood_counts_by_cond(mdata, test_var = 'stage_continuous', subset_nhoods = pl_nhoods)
``` 
![Screenshot 2022-11-03 at 12 39 54](https://user-images.githubusercontent.com/32264060/199722700-02be1f7c-6e49-45c3-b47b-a265b438c69c.png)

So the _subset_nhoods_ parameter is used to select the nhoods that you want to plot

> ```
> pl_df = pd.DataFrame(nhood_adata[subset_nhoods].X.A, columns=nhood_adata.var_names).melt(
>     var_name=nhood_adata.uns["sample_col"], value_name="n_cells"
> )
> pl_df = pd.merge(pl_df, nhood_adata.var)
> ```
> 
This bit of code makes a "long" dataframe from the cell abundance matrix (`nhood_adata.X`), then joins this dataframe with the covariate info for each sample stored in `nhood_adata.var`.

Does this clarify?

Reply:
Thanks for your clarification! It's now much clear to me.